### ISSUE TITLE
Shortest length fractions

### ISSUE THREAD

**Issue by lesobrod** [reallyasi9/EgyptianFractions.jl#8]
Hi! I did not find a function to calculate Shortest Length egyptian fractions. Can we make it together?

**Comment by reallyasi9** [comment]
Certainly. Are you proposing anything in particular? There is no algorithm that is better than polynomial time, and the best I could find is described in the answer https://mathoverflow.net/a/308386.

**Comment by scheinerman** [comment]
Just for fun, I created my own [Egyptian Fractions](https://github.com/scheinerman/MyEgyptianFractions.jl.git) Julia module in which I provide two methods for computing Egyptian Fraction representations (of positive rational numbers only). 

The first method is the greedy method (as described by Fibonnaci) and I'm sure is the same as your `efgreedy`.

The other is based on integer programming which works reasonably well if the numerator and denominator of the desired rational number are fairly small. The advantage is that it can sometimes find representations with smaller denominators. 

For example, 
```
julia> greedy(3//7)
3-element Vector{BigInt}:
   3
  11
 231

julia> optimal(3//7,30)
3-element Vector{Int64}:
  4
  7
 28
```

However, the method can give incorrect answers due to roundoff errors in the solvers. 

If you are interested in adding this method to this project, please let me know.

**Comment by reallyasi9** [comment]
Very clever, and pretty straight-forward to implement! I'd be happy to implement it here. Feel free to make a pull request of your own, too, if you can get to it before I do.

**Comment by scheinerman** [comment]
@reallyasi9 : Glad you like it. Though I get the general idea, I really don't understand how to fork and create pull requests. (I've occasionally received them.) I think you'll need to do these relatively easy steps:
* Copy the [source file](https://github.com/scheinerman/MyEgyptianFractions.jl/blob/main/src/Optim.jl) into your project.
* I suppose you'll want to rename the function to be `efoptimal` to be consistent with `efgreedy`.
* Add `using HiGHS`, `using JuMP`, and `using ChooseOptimizer` to the source file. 
* Add a test in `test/runtests.jl`.
* Update `Project.toml` to include these:
```
[deps]
ChooseOptimizer = "858a232f-1959-5553-8cfc-91e1fd5304e2"
HiGHS = "87dc4568-4c63-4d18-b0c0-bb2238e4078b"
JuMP = "4076af6c-e467-56ae-b986-b466b2749572"

[compat]
ChooseOptimizer = "0.3"
HiGHS = "1.14"
JuMP = "1.24"
```
